java/code/src/com/suse/manager/reactor/SaltReactor.java

```diff
@@ -326,7 +327,9 @@ else if (beaconEvent.getBeacon().equals("reboot_info")) {
             Optional<MinionServer> minion = MinionServerFactory.findByMinionId(beaconEvent.getMinionId());
             minion.ifPresent(
                 m -> {
-                    m.setRebootNeeded((Boolean) beaconEvent.getData().get("reboot_needed"));
+                    Boolean rebootRequired = (Boolean) beaconEvent.getData().get("reboot_needed");
+                    Date rebootRequiredAfter = rebootRequired ? new Date() : null;
```

Comment:
This unboxing might throw `NullPointerException`, if `reboot_needed` can be null. if so use `BooleanUtils.isTrue()`:
```suggestion
                    Date rebootRequiredAfter = BooleanUtils.isTrue(rebootRequired) ? new Date() : null;
```